chore: upgrade StyleLint version to 15

oguemon opened the pull request:
Upgrade StyleLint version from 14 to 15.

Ref: [Migrating to 15.0.0 | Stylelint](https://stylelint.io/migration-guide/to-15)

oguemon commented on styled-components.js:
```diff
@@ -11,12 +11,6 @@ module.exports = {
     // OK
     // background-image: url(${toStaticUrl('/img/foo.png')});
     "function-url-quotes": null,
-    // This rule cannot recognize CSS functions in template literals
-    // Error
-    // transform: ${({ rotate }) => `rotate(${rotate}deg)`};
-    // OK
-    // transform: ${({ rotate }) => `rotate(${rotate}deg) `};
-    "function-whitespace-after": null,
```
![memo](https://img.shields.io/badge/review-memo-white.svg)
`function-whitespace-after` is now deprecated and removed from `stylelint-config-standard`.

https://github.com/stylelint/stylelint-config-standard/compare/29.0.0...30.0.0#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346L93

oguemon commented on index.js:
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-  extends: ["stylelint-config-standard", "stylelint-config-prettier"],
```
![memo](https://img.shields.io/badge/review-memo-white.svg)
Ref: https://stylelint.io/migration-guide/to-15#deprecated-stylistic-rules.

oguemon commented on package.json:
```diff
@@ -15,7 +15,7 @@
   },
   "engines": {
-    "node": ">=12"
+    "node": ">=14.13.1"
```
![memo](https://img.shields.io/badge/review-memo-white.svg)
Removed support for Node.js 12.
https://stylelint.io/migration-guide/to-15#removed-support-for-nodejs-12

oguemon commented on styled-components.js:
```diff
@@ -1,6 +1,6 @@
 module.exports = {
   extends: ["./index.js", "stylelint-config-styled-components"],
-  customSyntax: "@stylelint/postcss-css-in-js",
```
![memo](https://img.shields.io/badge/review-memo-white.svg)
[@stylelint/postcss-css-in-js](https://www.npmjs.com/package/@stylelint/postcss-css-in-js) is deprecated, so I use [postcss-styled-syntax](https://www.npmjs.com/package/postcss-styled-syntax) alternatively.

https://stylelint.io/migration-guide/to-15#removed-processors-configuration-property

oguemon commented on styled-components.js:
```diff
@@ -1,6 +1,6 @@
 module.exports = {
   extends: ["./index.js", "stylelint-config-styled-components"],
-  customSyntax: "@stylelint/postcss-css-in-js",
+  customSyntax: "postcss-styled-syntax",
```
![memo](https://img.shields.io/badge/review-memo-white.svg)
[@stylelint/postcss-css-in-js](https://www.npmjs.com/package/@stylelint/postcss-css-in-js) is deprecated, so I use [postcss-styled-syntax](https://www.npmjs.com/package/postcss-styled-syntax) alternatively.

https://stylelint.io/migration-guide/to-15#removed-processors-configuration-property

oguemon commented on index.js:
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-  extends: ["stylelint-config-standard", "stylelint-config-prettier"],
+  extends: ["stylelint-config-standard"],
```
![memo](https://img.shields.io/badge/review-memo-white.svg)
Ref: https://stylelint.io/migration-guide/to-15#deprecated-stylistic-rules.

koba04 commented:
@oguemon 
Thank you!
Could you update the Node.js version on GitHub Actions? Let's remove `12.x` and add `18.x` instead.

https://github.com/cybozu/stylelint-config/blob/main/.github/workflows/test.yml#L17

koba04 reviewed (approved):
👍 LGTM!!! Great work 👍

koba04 commented:
@oguemon Could you fix the failing test?

oguemon commented:
@koba04
I fixed the failing test on Node.js 14! (Add Package https://github.com/cybozu/stylelint-config/pull/27/commits/10bc6c2f6bd49473bb6471030f046c1b9eb3ff04)

And I updated `actions/setup-node@v2` -> `@v3` (https://github.com/cybozu/stylelint-config/pull/27/commits/a90efef8b8054c53227b16275b025246968d8a07)

Prease check these changes!
